cli/tests/test_operations.rs

```diff
@@ -2609,6 +2609,105 @@ fn test_op_show() {
+    @  707335b2ab78 test-username@host.example.com 2001-02-03 04:05:12.000 +07:00 - 2001-02-03 04:05:12.000 +07:00
+    │  Recording a test transaction
+    │  args: jj op record 'Recording a test transaction'
+    ○  14a49257b1cb test-username@host.example.com 2001-02-03 04:05:10.000 +07:00 - 2001-02-03 04:05:10.000 +07:00
+    │  track remote bookmark bookmark-1@origin
+    │  args: jj bookmark track bookmark-1
+    ○  93de354b21ad test-username@host.example.com 2001-02-03 04:05:09.000 +07:00 - 2001-02-03 04:05:09.000 +07:00
+    │  fetch from git remote(s) origin
+    │  args: jj git fetch
+    ○  0ed329262b36 test-username@host.example.com 2001-02-03 04:05:08.000 +07:00 - 2001-02-03 04:05:08.000 +07:00
+    │  add git remote origin
+    │  args: jj git remote add origin ../git-repo
+    ○  8f47435a3990 test-username@host.example.com 2001-02-03 04:05:07.000 +07:00 - 2001-02-03 04:05:07.000 +07:00
+    │  add workspace 'default'
+    ○  000000000000 root()
+    [EOF]
+    "#);
+
+    // Add a transaction marking an older transaction
+    let output = work_dir.run_jj([
+        "op",
+        "record",
+        "--at-op",
+        "93de354b21ad",
+        "Recording a transaction with a specified parent",
+    ]);
+    insta::assert_snapshot!(output, @r"");
+
+    // Op log with new entry
+    let output = work_dir.run_jj(["op", "log"]);
+    insta::assert_snapshot!(output, @r#"
+    @    88858c7b8e68 test-username@host.example.com 2001-02-03 04:05:15.000 +07:00 - 2001-02-03 04:05:15.000 +07:00
+    ├─╮  reconcile divergent operations
+    │ │  args: jj op log
+    ○ │  707335b2ab78 test-username@host.example.com 2001-02-03 04:05:12.000 +07:00 - 2001-02-03 04:05:12.000 +07:00
+    │ │  Recording a test transaction
+    │ │  args: jj op record 'Recording a test transaction'
+    ○ │  14a49257b1cb test-username@host.example.com 2001-02-03 04:05:10.000 +07:00 - 2001-02-03 04:05:10.000 +07:00
+    │ │  track remote bookmark bookmark-1@origin
+    │ │  args: jj bookmark track bookmark-1
+    │ ○  1c8074d6827a test-username@host.example.com 2001-02-03 04:05:14.000 +07:00 - 2001-02-03 04:05:14.000 +07:00
+    ├─╯  Recording a transaction with a specified parent
+    │    args: jj op record --at-op 93de354b21ad 'Recording a transaction with a specified parent'
+    ○  93de354b21ad test-username@host.example.com 2001-02-03 04:05:09.000 +07:00 - 2001-02-03 04:05:09.000 +07:00
+    │  fetch from git remote(s) origin
+    │  args: jj git fetch
+    ○  0ed329262b36 test-username@host.example.com 2001-02-03 04:05:08.000 +07:00 - 2001-02-03 04:05:08.000 +07:00
+    │  add git remote origin
+    │  args: jj git remote add origin ../git-repo
+    ○  8f47435a3990 test-username@host.example.com 2001-02-03 04:05:07.000 +07:00 - 2001-02-03 04:05:07.000 +07:00
+    │  add workspace 'default'
+    ○  000000000000 root()
+    [EOF]
+    ------- stderr -------
+    Concurrent modification detected, resolving automatically.
+    [EOF]
+    "#);
```

Comment:
If we decide to go forward with this, I think we should add a test that modifies some files in the working copy and demonstrates whether `jj op waypoint` snapshots the working copy or not.

I'd guess it would (so it'd create two operations), and `jj op waypoint --ignore-working-copy` would not, which seems fine to me.